public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -354,20 +354,26 @@ export class PrometheusDatasource implements DataSourceApi<PromQuery> {
           })
           .value();
 
+        const dupCheck = {};
         for (const value of series.values) {
           const valueIsTrue = value[1] === '1'; // e.g. ALERTS
           if (valueIsTrue || annotation.useValueForTime) {
-            const event = {
+            const event: any = {
```

Comment:
seems unnecessary to define it as any & loose all typing, think we need an annotation event type, and add type definition for the annotationQuery method, like we have for query method. If you want me to do that I can try find time the next few days.

Reply:
I think so too, we need annotation event type. If you add it, really helpful for me.